apps/meteor/tests/e2e/federation/page-objects/fragments/home-content.ts

```diff
@@ -25,16 +23,72 @@ export class FederationHomeContent {
 		return this.page.locator('[data-qa-type="message"][data-sequential="false"]').last();
 	}
 
+	get typingIndicator(): Locator {
+		return this.page.locator('.rc-message-box__activity');
+	}
+
 	async sendMessage(text: string): Promise<void> {
+		await this.page.locator('[name="msg"]').type(text);
+		await this.page
+			.locator('//*[contains(@class, "rc-message-box__icon") and contains(@class, "rc-message-box__send") and contains(@class, "js-send")]')
+			.click();
+	}
+
+	async sendMessageUsingEnter(text: string): Promise<void> {
 		await this.page.locator('[name="msg"]').type(text);
 		await this.page.keyboard.press('Enter');
 	}
 
+	async editLastMessage(message: string): Promise<void> {
+		await this.openLastMessageMenu();
+		await this.btnOptionEditMessage.click();
+		await this.page.locator('[name="msg"]').fill(message);
+		await this.page.keyboard.press('Enter');
+	}
+
+	async deleteLastMessage(): Promise<void> {
+		await this.openLastMessageMenu();
+		await this.btnOptionDeleteMessage.click();
+		await this.page.locator('#modal-root dialog .rcx-modal__inner .rcx-modal__footer .rcx-button--danger').click();
+	}
+
+	async starLastMessage(): Promise<void> {
+		await this.openLastMessageMenu();
+		await this.btnOptionStarMessage.click();
+	}
+
+	async replyInDm(message: string): Promise<void> {
+		await this.openLastMessageMenu();
+		await this.btnOptionReplyDirectly.click();
+		await this.page.waitForTimeout(2000);
+		await this.page.locator('[name="msg"]').fill(message);
+		await this.page.keyboard.press('Enter');
+	}
+
+	async sendAudioRecordedMessage(): Promise<void> {
+		await this.btnRecordAudio.click();
+		await this.page.waitForTimeout(5000);
+		await this.page.locator('.rc-message-box__icon.rc-message-box__audio-message-done').click();
+		await this.btnModalConfirm.click();
+	}
+
+	async sendVideoRecordedMessage(): Promise<void> {
+		await this.btnMenuMoreActions.click();
+		await this.btnVideoMessage.click();
+		await this.page.locator('.vrec-dialog button.record').click();
+		await this.page.waitForTimeout(5000);
+		await this.page.locator('.vrec-dialog button.record').click();
+		await this.page.locator('.vrec-dialog button.ok').click();
+		await this.btnModalConfirm.click();
+	}
+
 	async dispatchSlashCommand(text: string): Promise<void> {
 		await this.page.locator('[name="msg"]').type(text);
+		await this.page.waitForTimeout(2000);
```

Comment:
Is it possible to change the _waitForTimeout_ for something more specific? So it's more clear what we are waiting  
e.g: waitForSelector, waitForEvent

```suggestion
		await this.page.locator('//*[contains(@class, "rc-message-box__icon") and contains(@class, "rc-message-box__send") and contains(@class, "js-send")]').waitFor()
```

Reply:
Thanks for the tip, I've removed it wherever and whenever it was possible, I just kept it in some specific cases.